libs/common/RespReadUtils.cs

```diff
@@ -624,14 +624,14 @@ public static bool ReadStringWithLengthHeader(out string result, ref byte* ptr,
         /// <summary>
         /// Read string with length header
         /// </summary>
-        public static bool ReadStringWithLengthHeader(MemoryPool<byte> pool, out MemoryResult<byte> result, ref byte* ptr, byte* end)
+        public static bool ReadStringWithLengthHeader(MemoryPool<byte> pool, out MemoryResult<byte> result, ref byte* ptr, byte* end, bool allowNull = false, bool isArray = false)
```

Comment:
I don't think `isArray` should ever be true when reading a string. The parameter is used to indicate whether we are reading a string length header or an array length header, but in this case we are always expecting a string length header.

Reply:
Remove isArray, and any other `if` checks that were added solely for the purpose of code reuse. This is inner loop so keeping minimal is preferred.

Reply:
I removed isArray where it was not needed and created a dedicated ReadStringResponseWithLengthHeader for the client